### ISSUE TITLE
Plotting distributions goes wrong?

### ISSUE THREAD

**Issue by nilshg** [JuliaPlots/StatsPlots.jl#245]
```
using StatsPlots, Distributions
density(Normal())
```
currently gives a weird bimodal distribution

![normal_error](https://user-images.githubusercontent.com/7264491/60345293-997e5d00-99b0-11e9-8d19-ba28c3e28926.PNG)

This is on Julia v1.1.1

```
    Status `C:\Users\~\.julia\environments\v1.1\Project.toml`
  [c52e3926] Atom v0.8.7
  [6e4b80f9] BenchmarkTools v0.4.2
  [336ed68f] CSV v0.5.6
  [a93c6f00] DataFrames v0.18.3
  [31c24e10] Distributions v0.19.2
  [38e38edf] GLM v1.2.0
  [7073ff75] IJulia v1.18.1
  [e5e0dc1b] Juno v0.7.0
  [ff71e718] MixedModels v1.1.6
  [c030b06c] ODE v2.4.0
  [91a5bcdd] Plots v0.25.1
  [6f49c342] RCall v0.13.3
  [d519eb52] RegressionTables v0.2.0
  [2913bbd2] StatsBase v0.30.0
  [f3b207a7] StatsPlots v0.11.0
```

**Comment by nico202** [comment]
I can reproduce.
```
Julia Version 1.1.0
Commit 80516ca202* (2019-01-21 21:24 UTC)
Platform Info:
  OS: Linux (x86_64-unknown-linux-gnu)
  CPU: Intel(R) Core(TM)2 CPU         P8600  @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, penryn)
```

```
    Status `~/.julia/environments/v1.1/Project.toml`
  [31c24e10] Distributions v0.19.2
  [91a5bcdd] Plots v0.25.2
  [f3b207a7] StatsPlots v0.11.0
```

**Comment by daschw** [comment]
I think
```julia
plot(Normal())
```
does what you want.
![normal](https://user-images.githubusercontent.com/16589944/60395678-61555680-9b37-11e9-8ed1-2b1eee039a78.png)

`density` tries to fit a kde based on sample data, like e.g.
```julia
data = rand(Normal(), 100)
density(data)
```
![density](https://user-images.githubusercontent.com/16589944/60395682-69ad9180-9b37-11e9-8d14-55cf359cc26d.png)

**Comment by mkborregaard** [comment]
It'd be more ideomatic to have `density(Normal())` work like `plot` though, wouldn't it?

**Comment by daschw** [comment]
Yes, but I think then we have to change the type recipe for distributions. I have not thought about it in detail, though.

**Comment by nilshg** [comment]
Okay makes sense, but should it maybe error rather than plotting this bimodal distribution? Where does it come from?

**Comment by Moelf** [comment]
MethodError: no method matching Float64(::Array{Float64,1})

from
```
plot(Normal())
```

**Comment by alecloudenback** [comment]
I agree that when asking for the `density` of a distribution, it makes a lot more sense to have something like what `plot` returns. I tried to follow how the above bimodal curve is derived, but debugging in Juno just takes me to [RecpiesBase](https://github.com/JuliaPlots/RecipesBase.jl/blob/c92e56a5ad46a50a988d616f906b43f7976cc681/src/RecipesBase.jl#L389) and the debug seems to not be able to continue into the macro. Is this a RecipesBase issue or a StatsPlots issue?

**Comment by sethaxen** [comment]
The user recipe for a `d::Distribution` converts it to `x` a set of x-values and `y` to a set of density values at `x`. But the `density` recipe discards `x` and interprets `y` as a random sample, so `density(Normal())` is plotting the distribution of PDF values of `Normal()`.

This is easy to fix by checking in the user recipe for `Distribution` if `seriestype` is set to `:density`, and if it is, switching it to `:path`.

**Comment by newptcai** [comment]
I have a student asking me about it today. I have to say I was quite confused.
